docs/workflow/pycbc_make_coinc_search_workflow.rst

```diff
@@ -608,53 +608,250 @@ If you have a subdax that failed, ``pegasus_analyzer`` will provide you with a c
+
+----------------------------
+Re-running a failed workflow
+----------------------------
+
+Ocasionaly it may be necessary to use data from a partially completed
+workflow, e.g. if there a bug in an executable and you wish to re-run the
+workflow with a new version of the executable. If the workflow failed, no
+results web page will have been generated and the output data may not have
+been copied to the locations in ``main.map``. To re-use data from a previous
+failed workflow, you need to create a cache file containing the completed jobs
+from the previous workflow. 
+
+To do this, ``cd`` into the ``local-site-scratch/work`` directory of your
+failed workflow. For example, if you used ``--output-dir output`` when
+planning the workflow, the then run the command::
+
+    cd /path/to/workflow/output/local-site-scratch/work
+
+Once in this directory there should be a directory that ends with
+``main_ID0000001`` (e.g. ``my-workflow-main_ID0000001``) Change into that
+directory.
+
+Once in the ``main_ID0000001`` directory, run the command::
+
+    for pfn in `find . -type f | sed 's+^./++g'` ; do echo $pfn file://`pwd`/$pfn pool=\"local\" ; done > /path/to/partial_workflow.map
+
+
+changing ``/path/to`` to a location where you want to save the cache.
+
+If your run used the Open Science Grid, you will need to do the same thing for
+the ``osg-site-scratch`` directory. In this case::
+
+    cd /path/to/workflow/output/osg-site-scratch/work
+
+and change into the directory that ends with ``main_ID0000001``. Then run the
+same command as above::
+
+    for pfn in `find . -type f | sed 's+^./++g'` ; do echo $pfn file://`pwd`/$pfn pool=\"local\" ; done >> /path/to/partial_workflow.map
+
+but **note** the ``>>`` rather than ``>`` to append to the file ``partial_workflow.map``, rather than clobbering the file and creating a new file.
+ 
+Now you can than use the ``partial_workflow.map`` cache file as the ``--cache-file`` argument to ``pycbc_submit_dax``.
+
+
+-----------------------------------------------
+Using partial products from a previous workflow
+-----------------------------------------------
+
+If you are changing the configuration parameters of a workflow, then you can
+build a cache file from a previous ``main.map`` file or the files under
+``local-site-scratch``, but you will need to filter the cache file to remove
+the files for jobs that have a changed configuration.  Here are a few
+examples:
+
+* If you are changing the configuration of ``pycbc_inspiral`` you must regenerate almost all the files in the workflow so it easier to start from scratch.
+* If you are changing the injections, but want to re-use the ``FULL_DATA`` previous analysis, you can filter the ``main.map`` to keep the veto files, template bank files, full data inspiral files, and PSD files. For example::
+
+    egrep '(VETO|BANK|FULL_DATA|PSD)' /path/to/main.map > /path/to/reuse.map
```

Comment:
will this pick up some full data '.dax' files?  I have not found that reuse works properly if .dax files are in the cache